docs/sources/flow/reference/components/pyroscope.ebpf.md

```diff
@@ -0,0 +1,250 @@
+|---------------------------|--------------------------|--------------------------------------------------------------|---------|----------|
+| `forward_to`              | `list(ProfilesReceiver)` | List of receivers to send collected profiles to.             |         | yes      |  
+| `targets`                 | `list(map(string))`      | List of targets to group profiles by container id            |         | yes      |   
+| `collect_interval`        | `duration`               | How frequently to collect profiles                           | `15s`   | no       |       
+| `sample_rate`             | `int`                    | How many times per second to collect profile samples         | 97      | no       |     
+| `pid_cache_size`          | `int`                    | The size of the pid -> proc symbols table LRU cache          | 32      | no       |      
+| `build_id_cache_size`     | `int`                    | The size of the elf file build id -> symbols table LRU cache | 64      | no       |       
+| `same_file_cache_size`    | `int`                    | The size of the elf file -> symbols table LRU cache          | 8       | no       |       
+| `container_id_cache_size` | `int`                    | The size of the pid -> container ID table LRU cache          | 1024    | no       |       
+| `collect_user_profile`    | `bool`                   | A flag to enable/disable collection of userspace profiles    | true    | no       |       
+| `collect_kernel_profile`  | `bool`                   | A flag to enable/disable collection of kernelspace profiles  | true    | no       |       
+
+## Exported fields
+
+`pyroscope.ebpf` does not export any fields that can be referenced by other
+components.
+
+## Component health
+
+`pyroscope.ebpf` is only reported as unhealthy if given an invalid
+configuration.
+
+## Debug information
+
+* `targets` currently tracked active targets.
+* `pid_cache` per process elf symbol tables and their sizes in symbols count.
+* `elf_cache` per build id and per same file symbol tables and their sizes in symbols count.
+
+## Debug metrics
+
+* `pyroscope_fanout_latency` (histogram): Write latency for sending to direct and indirect components.
+* `pyroscope_ebpf_active_targets` (gauge): Number of active targets tracked by the component.
+* `pyroscope_ebpf_profiling_sessions_total` (counter): Number of profiling sessions completed.
+* `pyroscope_ebpf_profiling_sessions_failing_total` (counter): Number of profiling sessions failed.
+* `pyroscope_ebpf_pprofs_total` (counter): Number of pprof profiles collected by ebpf component.
+
+## Profile collecting behavior
+
+The `pyroscope.ebpf` component is designed to collect stacktraces associated with a process running on the current host.
+Stack traces are collected according to the defined `sample_rate`, meaning traces are gathered this many times
+per second.
```

Comment:
`sample_rate` is the profiler argument which has default value so it may be defined by the user.

> The sample rate is measured in traces collected per second.
The sample rate is not measured.
If user specified `sample_rate` argument as 239, then stack traces are collected 239 times per second
If user does not specify `sample_rate`, then default value of 97 is used and stack traces are collected 97 times per second.